includes/class-wc-stripe-intent-controller.php

```diff
@@ -299,12 +299,14 @@ public function create_payment_intent( $order_id = null ) {
 			$amount = $order->get_total();
 		}
 
+		$gateway = new WC_Stripe_UPE_Payment_Gateway();
```

Comment:
I was debating in my head whether the ajax hook should be in the `WC_Stripe_Intent_Controller` and then instantiate the UPE gateway or whether the hook should be in the UPE gateway and then the Intent Controller should be instantiated instead. Personally I prefer the latter, though I don't really have much reasoning to say the other approach is more wrong.

However, if we are using an instantiation of the gateway here, there is a `get_gateway` function in this `WC_Stripe_Intent_Controller` class that perhaps we could use here? Though, that function would need to be adapted slightly. The purpose of that `get_gateway` function appears to be to select the correct gateway when Subscriptions may be present; in WCPay we resolved this by turning the Subscriptions compat class into a trait and adding it to the gateway. If we choose to do this, then most of what I've described will be redundant anyways.

Reply:
The current stripe gateways don't contain the intent hooks, they are in this `WC_Stripe_Intent_Controller`, that's why I put the new UPE hooks here. 

But I agree that the WC Payments approach is clearer; maybe we can create an issue to consolidate all hooks in the UPE gateway when we add support for UPE subscriptions (in case we end up using this intent class after all 🤷🏼 )? Or do you think we should do this in this PR?

Reply:
Cool, good point. I don't mind saving this dilemma for another day: it makes sense to me to leave this one until we sort out subscriptions as well. Let's not hold back the glorious march of progress in this PR for this little quibble.